packages/engine/src/networking/serialization/DataReader.ts

```diff
@@ -78,10 +80,83 @@ export const readVector4 = (vector4: Vector4SoA) => (v: ViewCursor, entity: Enti
   if (checkBitflag(changeMask, 1 << b++)) readComponentProp(v, vector4.w, entity)
 }
 
+// Reads a compressed rotation value from the network stream. This value must have been previously written
+// with WriteCompressedRotation() in order to be properly decompressed.
+export const readCompressedRotation = (vector4: Vector4SoA) => (v: ViewCursor, entity: Entity | undefined) => {
+  const changeMask = readUint8(v)
+  if (changeMask <= 0) return
+
+  let compressedBinaryData = readUint32(v)
+
+  const expand = (compressedBinaryData: number) => {
+    const valueReadMask = 0b00000000000000000000000111111111
```

Comment:
This function is being recreated every time readCompressedRotation is called, which creates unnecessary garbage and slows down performance.